usecases/backup/zip.go

```diff
@@ -33,37 +34,76 @@ import (
 type CompressionLevel int
 
 const (
-	DefaultCompression CompressionLevel = iota
-	BestSpeed
-	BestCompression
+	GzipDefaultCompression CompressionLevel = iota
+	GzipBestSpeed
+	GzipBestCompression
+	ZstdBestSpeed
+	ZstdDefaultCompression
+	ZstdBestCompression
+	NoCompression
 )
 
+type compressor interface {
+	Flush() error
+	Write(p []byte) (n int, err error)
+	Close() error
+}
+
 type zip struct {
-	sourcePath string
-	w          *tar.Writer
-	gzw        *gzip.Writer
-	pipeWriter *io.PipeWriter
-	counter    func() int64
+	sourcePath       string
+	w                *tar.Writer
+	compressorWriter compressor
+	pipeWriter       *io.PipeWriter
+	counter          func() int64
 }
 
-func NewZip(sourcePath string, level int) (zip, io.ReadCloser) {
+func NewZip(sourcePath string, level int) (zip, io.ReadCloser, error) {
```

Comment:
nit: we could rename this method to `NewCompressor`, this class now supports more than one compression type